### PR TITLE
feat: add ibcToken contract

### PR DESCRIPTION
**Pull request by 0xpatrickdev**

### CODE REVIEW

File: contract/src/ibcToken.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    getTokenKit: M.call().returns(M.any()),
+  }),
+);
+
+/**
+ * @param {ZCF<IBCTokenTerms>} zcf
+ * @param {Record<string, unknown>} _privateArgs
+ * @param {import('@agoric/vat-data').Baggage} baggage
+ */
+export const prepare = async (zcf, _privateArgs, baggage) => {
```
Comment (dckc) [inline contract/src/ibcToken.js:39]: Have you seen mintHolder.js in agoric-sdk ? Does the same thing, afaict

Comment (0xpatrickdev) [reply contract/src/ibcToken.js:39]: I saw it but did not look closely enough so thank you for the pointer.

I suppose we also do not need to produce an installation or instance - just a `myTokenKit` that holds the creator/mint facet.

File: contract/src/start-ibcToken.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+ *  ibcToken: {
+ *    bundleID: string;
+ *  };
+ *  tokenKeyword: string;
+ *  tokenDecimals: number;
+ *  proposedName: string;
+ *  denom: string;
+ * }}} config
+ */
+export const startIBCToken = async (powers, config) => {
```
Comment (dckc) [inline contract/src/start-ibcToken.js:78]: Aside from the `boardAux` stuff, this is in the SDK too: [publishInterchainAssetFromBank](https://github.com/Agoric/agoric-sdk/blob/mainnet1B-rc3/packages/inter-protocol/src/proposals/addAssetToVault.js#L55)

**Comment by dckc** [comment]
This stuff is already on chain.

So all you need is something like [add-stATOM.js](https://github.com/0xpatrickdev/agoric-vault-collateral-proposal/blob/main/add-stATOM.js) but
 1. prune `overrideManifest` down to just `publishInterchainAssetFromBank`, and
 2. open-code the `boardAux` stuff as in `start-ibcToken.js` here. Note that importing marshal isn't practical in this context, so it may be better to use a fill-in-the-template approach.

File: contract/src/ibcToken.js (hunk 1 above)
Comment (dckc) [reply contract/src/ibcToken.js:39]: using `mintHolder.js` does involve starting a new instance of it

File: contract/src/start-ibcToken.js [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+ * @file core eval script* to start the ibcToken contract.
+ *
+ * * see test-gimix-proposal.js to make a script from this file.
+ *
+ * The `permit` export specifies the corresponding permit.
+ */
+// @ts-check
+
+import { E } from '@endo/far';
+import { makeMarshal } from '@endo/marshal';
```
Comment (dckc) [inline contract/src/start-ibcToken.js:11]: The gimix approach to core-eval proposals doesn't actually link imports. It erases import declarations, assuming they're for stuff like `E` that's provided by the core-eval compartment. So any calls to `makeMarshal` won't work.

**Pull request closed without merging**